src/D2L.Bmx/Aws/AwsCredsCache.cs

```diff
@@ -0,0 +1,83 @@
+using System.Runtime.InteropServices;
+using System.Text.Json;
+namespace D2L.Bmx.Aws;
+
+internal interface IAwsCredentialCache {
+	public AwsCredentials? GetCredentials( string org, string user, AwsRole role, int duration );
+	public void SetCredentials( string org, string user, AwsRole role, AwsCredentials credentials );
+}
+
+internal class AwsCredsCache : IAwsCredentialCache {
+	private static void WriteTextToFile( string path, string content ) {
+		var op = new FileStreamOptions {
+			Mode = FileMode.Create,
+			Access = FileAccess.Write,
+		};
+		if( !RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			op.UnixCreateMode = UnixFileMode.UserRead | UnixFileMode.UserWrite;
+		}
+		using var writer = new StreamWriter( path, op );
+		writer.Write( content );
+	}
+
+	private List<AwsCacheModel> GetAllCache() {
+		if( !File.Exists( BmxPaths.AWS_CREDS_CACHE_FILE_NAME ) ) {
+			return new();
+		}
+		try {
+			string sessionsJson = File.ReadAllText( BmxPaths.AWS_CREDS_CACHE_FILE_NAME );
+			return JsonSerializer.Deserialize( sessionsJson, SourceGenerationContext.Default.ListAwsCacheModel )
+				?? new();
+		} catch( JsonException ) {
+			return new();
+		}
+	}
+
+	void IAwsCredentialCache.SetCredentials( string org, string user, AwsRole role, AwsCredentials credentials ) {
+		List<AwsCacheModel> allEntries = GetAllCache();
+
+		allEntries.Add( new AwsCacheModel(
+			Org: org,
+			User: user,
+			RoleArn: role.RoleArn,
+			Credentials: credentials
+		) );
+
+		var ttlLongerThan = DateTime.UtcNow.AddMinutes( 10 );
+
+		var prunedEntries = allEntries
+			// Remove expired/expiring & entries not related to current user
+			.Where( o =>
+				o.Credentials.Expiration >= ttlLongerThan
```

Comment:
nit: personally would prefer wording like `minimumTtl` or `minimumExpiryTime`. I had trouble understanding what `ttlLongerThan` meant what I read the definition above.